mlperf_logging/result_summarizer/result_summarizer.py

```diff
@@ -331,19 +570,40 @@ def main():
             orgs = multiple_folders.group(2).split(',')
         elif '*' == path_suffix:
             orgs = os.listdir(path_prefix)
-            orgs = [org for org in orgs
-                    if _is_organization_folder(os.path.join(path_prefix, org))]
+            orgs = [
+                org for org in orgs
+                if _is_organization_folder(os.path.join(path_prefix, org))
+            ]
         print('Detected organizations: {}'.format(', '.join(orgs)))
         for org in orgs:
             org_folder = path_prefix + org
-            summarize_results(org_folder, args.usage, args.ruleset, csv_file)
+            _update_summaries(org_folder)
     else:
         # Parse results for single organization.
-        summarize_results(args.folder, args.usage, args.ruleset, csv_file)
-    
-    # Close csv file if required
-    if args.csv is not None:
-        csv_file.close()
+        _update_summaries(args.folder)
+
+    # Print and write back results.
+    def _print_and_write(summaries, weak_scaling=False, mode='w'):
+        if len(summaries) > 0:
+            summaries = pd.concat(summaries).astype(
+                _get_column_schema(
+                    args.usage,
+                    args.ruleset,
+                    weak_scaling=weak_scaling,
+                ))
+            if weak_scaling:
+                print('Weak Scaling Scores:')
+            else:
+                print('Strong Scaling Scores:')
+            print(summaries)
+            if args.csv is not None:
+                summaries.to_csv(args.csv, index=False, mode=mode)
+
+    with pd.option_context('display.max_rows', None, 'display.max_columns',
```

Comment:
Have the summaries changed format?

Reply:
Yes. Without the `--csv` flag, it looks like:
```bash
(mlperf-logging) shangw@shangw-dt:~/workspace/logging$ python -m mlperf_logging.result_summarizer ../training_results_v1.0/Google/ training 1.0.0
Failed checks for eval_accuracy :  v['value'] >= 0.908
Compliance check failed in ../training_results_v1.0/Google/results/tpu-v4-1024-JAX/unet3d/result_1.txt
Failed checks for eval_accuracy :  v['value'] >= 0.908
Compliance check failed in ../training_results_v1.0/Google/results/tpu-v4-1024-JAX/unet3d/result_10.txt
Failed checks for eval_accuracy :  v['value'] >= 0.908
Compliance check failed in ../training_results_v1.0/Google/results/tpu-v4-1024-JAX/unet3d/result_35.txt
Failed checks for eval_accuracy :  v['value'] >= 0.908
Compliance check failed in ../training_results_v1.0/Google/results/tpu-v4-1024-JAX/unet3d/result_7.txt
NOTICE: Dropping non-converged run(s) for Google tpu-v4-1024-JAX/unet3d using olympic scoring.
Strong Scaling Scores:
  division availability submitter       system host_processor_model_name  \
0   closed         None    Google  tpu-v4-1024                  AMD Rome
1   closed         None    Google   tpu-v4-128                  AMD Rome
2   closed         None    Google   tpu-v4-128                  AMD Rome
3   closed         None    Google  tpu-v4-2048                  AMD Rome
4   closed         None    Google   tpu-v4-256                  AMD Rome
5   closed         None    Google  tpu-v4-4096                  AMD Rome
6   closed         None    Google  tpu-v4-6912                  AMD Rome
7     open         None    Google   tpu-v4-256                  AMD Rome

   host_processors_count accelerator_model_name  accelerators_count  \
0                    256                    TPU                 512
1                     32                    TPU                  64
2                     32                    TPU                  64
3                    512                    TPU                1024
4                     64                    TPU                 128
5                   1024                    TPU                2048
6                   1728                    TPU                3456
7                     64                    TPU                 128

    framework      bert      dlrm   maskrcnn  minigo    resnet       ssd  \
0         JAX       NaN       NaN        NaN     NaN       NaN       NaN
1         JAX       NaN       NaN        NaN     NaN       NaN       NaN
2  TensorFlow  4.058681  0.843044  15.226350     NaN  4.301094  1.546778
3  TensorFlow       NaN       NaN   4.683094     NaN       NaN       NaN
4  TensorFlow       NaN  0.641072        NaN     NaN       NaN       NaN
5  TensorFlow  0.317275       NaN        NaN     NaN  0.284839  0.337861
6  TensorFlow  0.276035       NaN        NaN     NaN  0.229039       NaN
7         JAX       NaN       NaN        NaN     NaN  3.318239       NaN

   rnnt     unet3d  \
0   NaN   6.165925
1   NaN  27.516964
2   NaN        NaN
3   NaN        NaN
4   NaN        NaN
5   NaN        NaN
6   NaN        NaN
7   NaN        NaN

                                                                                      details_url  \
0  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-1024.json
1   https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-128.json
2   https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-128.json
3  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-2048.json
4   https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-256.json
5  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-4096.json
6  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-6912.json
7   https://github.com/mlcommons/training_results_v1.0/blob/master/Google/systems/tpu-v4-256.json

                                                                           code_url
0  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
1  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
2  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
3  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
4  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
5  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
6  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
7  https://github.com/mlcommons/training_results_v1.0/blob/master/Google/benchmarks
```
In my opinion, this looks a lot more human-friendly than csv. Of course you could still get the csv with `--csv`.

HPC actually has a slight problem with csv. HPC would need to produce two tables, one for strong scaling, the other for weak scaling. Currently I'm dumping both csvs into the same file, but I'm not sure if this format is readable by excel or other common csv readers.